### DISCUSSION TITLE
Vercel build outputs more than 15k files

### DISCUSSION THREAD

**Discussion by martin-dimi** [vercel/vercel#11018] [Help] [answered]
### Question

I've recently started hitting:
```
Vercel CLI 33.0.2
Error: Invalid request: `files` should NOT have more than 15000 items, received 16508.
```
after updating to next v14 (from 13.4). 

After some investigation, I've found that node_modules is included under `.vercel/output/functions`, specifically:
```
| path              | # files |
|-------------------|---------|
| .                 | 16092   |
| ./settings.func   | 5162    |
| ./api             | 5558    |
| ./_not-found.func | 5327    |
| ./src             | 3       |
... so on and so forth
```

For instance, `_not-found.func` has:
```
| path           | # files |
|----------------|---------|
| .              | 5327    |
| ./frontend     | 559     |
| ./node_modules | 4764    |
| ./common       | 2       |
```

Is it normal to see `node_modules` included in these files? 
The actual error occurs when I do `vercel deploy --prebuilt`. I've got the following `.vercelignore` file:
```
node_modules // have tried **/node_modules too
dist
.cache
```

PS: i'm running a monorepo, where `frontend` is a nextjs project (the one i'm deploying)

**Comment by martin-dimi** [comment]
Bump 🙏

**Comment by swarnava** [comment, marked as answer]
Can you try deploying using `vercel deploy --prebuilt --archive=tgz` ?

**Reply by martin-dimi** [reply]
That's what we've had to do, yes. Is it normal to push raw node_modules however? 
It comes up to be 1GB..
